chunkset_tpl.h

```diff
@@ -17,10 +17,10 @@ Z_INTERNAL uint32_t CHUNKSIZE(void) {
    (chunk_t bytes or fewer) will fall straight through the loop
    without iteration, which will hopefully make the branch prediction more
    reliable. */
-Z_INTERNAL uint8_t* CHUNKCOPY(uint8_t *out, uint8_t const *from, unsigned len) {
+Z_INTERNAL uint8_t* CHUNKCOPY(uint8_t *out, uint8_t const *from, uint32_t len) {
     Assert(len > 0, "chunkcopy should never have a length 0");
     chunk_t chunk;
-    int32_t align = ((len - 1) % sizeof(chunk_t)) + 1;
+    int32_t align = len % sizeof(chunk_t);
     loadchunk(from, &chunk);
     storechunk(out, &chunk);
     out += align;
```

Comment:
When you are using simple module, `align` can be `0`which means this code block should be protected or it will copy exactly same data twice.

Reply:
There is already an Assert to check to make sure Len is not zero

Reply:
How will it copy same data twice? If align is zero it will do the copy then subtract align from Len and Len will be zero and won't go into the whole loop to copy again.

Reply:
Because it doesn't update `out` correctly if `len` is exact multiple of chunk size.

Reply:
Actually I see what you mean. Only in case of sizeof(chunk_t) will it copy twice.

Reply:
Range of `align` is 0...sizeof(chunk_t)-1... as such every multiple of sizeof(chunk_t) causes first chunk to be copied twice.

Reply:
I'm not sure why I had changed that line. Probably a bad cherry-pick, because I had already addressed something similar in another PR. Anyways, it is resolved now.